### ISSUE TITLE
Tabstrip background is incorrect for some themes

### ISSUE THREAD

**Issue by zenparsing** [brave/brave-browser#35905]
## Description 

The tabstrip background color (in horizontal tabs mode) is incorrect for some themes.

## Steps to Reproduce

   1. Install a theme like [this one](https://chromewebstore.google.com/detail/elementary-freya-dark/lcmjgikfofdpfddimlpihnncnpdmlppl)

## Actual result:

<img src="https://private-user-images.githubusercontent.com/8007967/303159001-6b940023-6e5a-4bb8-af2b-327d239480f2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDc0MjU2MjcsIm5iZiI6MTcwNzQyNTMyNywicGF0aCI6Ii84MDA3OTY3LzMwMzE1OTAwMS02Yjk0MDAyMy02ZTVhLTRiYjgtYWYyYi0zMjdkMjM5NDgwZjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDhUMjA0ODQ3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTRlZTg1ODI4MDE3N2UxMGUxNjExZWY1YzcxZWM4M2Y4ZTAwNTdmNWJjNzdiNmVlNGQ5MmY1NWRkMTY5ZWIyNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.IU3PgfDTsXBCoYtnYQ82gu9TaVC-4f6edgvzZyuusMI" />

## Expected result:

<img width="1159" alt="Screenshot 2024-02-08 at 3 46 48 PM" src="https://github.com/brave/brave-browser/assets/5995084/6f592502-e80e-4029-a4e2-ef3a6fce849d">

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome?

**Comment by kjozwiak** [comment]
The above requires `1.63.161` or higher for `1.63.x` verification 👍

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.63.161 Chromium: 122.0.6261.57 (Official Build) (64-bit)
-- | --
Revision | 37b02b4a373f40114e70cc034e491bcc8380f23a
OS | Windows 10 Version 22H2 (Build 19045.4046)
```
- The issue is reproduced in 1.62.x and ensured that the issue is fixed on 1.63.161

**`1.62.165`** | **`1.63.161`**
---------------|----------------
![image](https://github.com/brave/brave-browser/assets/38657976/9b1bbd42-e10a-4210-a164-b2f659012ade) | ![image](https://github.com/brave/brave-browser/assets/38657976/e4e9248e-6279-454e-a55c-b16a80363910)
